adobe_vipm/flows/helpers.py

```diff
@@ -263,14 +296,55 @@ def get_licenses_and_consumables_count(subscriptions):
         """
         count_licenses = 0
         count_consumables = 0
-        for subscription in subscriptions["items"]:
-            if is_consumables_sku(subscription["offerId"]):
-                count_consumables += subscription["currentQuantity"]
-            else:
-                count_licenses += subscription["currentQuantity"]
 
+        if subscriptions and "items" in subscriptions:
+            for subscription in subscriptions["items"]:
+                sku = get_adobe_product_by_marketplace_sku(subscription["offerId"][0:10])
+                if (sku.is_valid_3yc_type() and
+                    subscription["autoRenewal"]["enabled"]):
+                    if sku.is_consumable():
+                        count_consumables += subscription["autoRenewal"]["renewalQuantity"]
+                    else:
+                        count_licenses += subscription["autoRenewal"]["renewalQuantity"]
+
+        return count_licenses, count_consumables
+
+    @staticmethod
+    def process_donwsize_lines_quantities(context, count_licenses, count_consumables):
+        for line in context.downsize_lines:
+            delta = line["oldQuantity"] - line["quantity"]
+            sku = get_adobe_product_by_marketplace_sku(line["item"]["externalIds"]["vendor"])
+            if sku.is_valid_3yc_type():
+                if sku.is_consumable():
+                    count_consumables -= delta
+                else:
+                    count_licenses -= delta
+        return count_licenses, count_consumables
+
+    @staticmethod
+    def process_upsize_lines_quantities(context, count_licenses=0, count_consumables=0):
+        for line in context.upsize_lines + context.new_lines:
+            delta = line["quantity"] - line["oldQuantity"]
+            sku = get_adobe_product_by_marketplace_sku(line["item"]["externalIds"]["vendor"])
+            if sku.is_valid_3yc_type():
+                if sku.is_consumable():
+                    count_consumables += delta
+                else:
+                    count_licenses += delta
         return count_licenses, count_consumables
 
+    @staticmethod
+    def validate_items_in_subscriptions(context, subscriptions):
+        if subscriptions["items"]:
```

Comment:
one more `if` here... Probably it is time to do in in the main function ones? To have `subscriptions` variables eiuther the list of subscriptions or `[]`?